tests/integ/test_reduce_describe_query.py

```diff
@@ -248,6 +248,19 @@ def test_select_quoted_identifiers(
         assert quoted_identifiers == expected_quoted_identifiers
 
 
+def test_snowflake_values(session):
+    df = session.create_dataframe([[1, 2], [3, 4]], schema=["a", "b"])
+    expected_quoted_identifiers = ['"A"', '"B"']
+    if session.reduce_describe_query_enabled:
+        with SqlCounter(query_count=0, describe_count=0):
+            assert df._plan._metadata.quoted_identifiers == expected_quoted_identifiers
+            assert df._plan.quoted_identifiers == expected_quoted_identifiers
```

Comment:
df._plan.attributes will still trigger a describe query because we call select() here so data type is lost currently. We can improve later. https://github.com/snowflakedb/snowpark-python/blob/072c8cfa9a78a9a2fd4a49af0b0082601fc34c4d/src/snowflake/snowpark/session.py#L3048